File: internal/handler/webhooks/hook_execution_notice.go
package webhooks

import (
	"fmt"

	"github.com/gin-gonic/gin"
)

// HookExecutionNotice handles setting flash notices based on webhook execution results
type HookExecutionNotice struct{}

// NewHookExecutionNotice creates a new instance of HookExecutionNotice
func NewHookExecutionNotice() *HookExecutionNotice {
	return &HookExecutionNotice{}
}

// SetHookExecutionNotice sets a flash notice based on the webhook execution result
func (h *HookExecutionNotice) SetHookExecutionNotice(ctx *gin.Context, result map[string]interface{}) {
	// Extract HTTP status and message from the result
	httpStatus, hasStatus := result["http_status"].(int)
	message, hasMessage := result["message"].(string)

	if !hasMessage {
		message = "Unknown error"
	}

	// Set appropriate flash notice based on HTTP status
	if hasStatus && httpStatus >= 200 && httpStatus < 400 {
		// Success case
		ctx.Set("flash_notice", fmt.Sprintf("Hook executed successfully: HTTP %d", httpStatus))
	} else if hasStatus {
		// HTTP error case
		ctx.Set("flash_alert", fmt.Sprintf("Hook executed successfully but returned HTTP %d %s", httpStatus, message))
	} else {
		// Execution failure case
		ctx.Set("flash_alert", fmt.Sprintf("Hook execution failed: %s", message))
	}
}

File: rust_app/src/services/webhooks/hook_execution_notice.rs
use actix_web::HttpResponse;
use crate::webhooks::WebHookResult;

pub trait HookExecutionNotice {
    fn set_hook_execution_notice(&self, result: &WebHookResult) -> HttpResponse {
        let http_status = result.http_status;
        let message = &result.message;

        if let Some(status) = http_status {
            if status >= 200 && status < 400 {
                HttpResponse::Ok().json(json!({
                    "notice": format!("Hook executed successfully: HTTP {}", status)
                }))
            } else {
                HttpResponse::Ok().json(json!({
                    "alert": format!("Hook executed successfully but returned HTTP {} {}", status, message)
                }))
            }
        } else {
            HttpResponse::Ok().json(json!({
                "alert": format!("Hook execution failed: {}", message)
            }))
        }
    }
} 